e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs

```diff
@@ -667,6 +648,318 @@ private static PrivateKeyFile GetPrivateSshKey(IIoTPlatformTestContext context)
+        }
+
+        /// <summary>
+        /// Gets the application ID associated with the DiscoveryUrl property of <paramref name="context"/>
+        /// </summary>
+        /// <param name="context">Shared Context for E2E testing Industrial IoT Platform</param>
+        /// <param name="ct">Cancellation token</param>
+        public static async Task<string> Registry_GetApplicationIdAsync(
+                IIoTPlatformTestContext context,
+                CancellationToken ct = default) {
+
+            var accessToken = await GetTokenAsync(context, ct).ConfigureAwait(false);
+
+            var client = new RestClient(context.IIoTPlatformConfigHubConfig.BaseUrl) { Timeout = TestConstants.DefaultTimeoutInMilliseconds };
+
+            var request = new RestRequest(Method.GET);
+            request.AddHeader(TestConstants.HttpHeaderNames.Authorization, accessToken);
+            request.Resource = TestConstants.APIRoutes.RegistryApplications;
+
+            var response = await client.ExecuteAsync(request, ct).ConfigureAwait(false);
+            Assert.NotNull(response);
+
+            if (!response.IsSuccessful) {
+                context.OutputHelper.WriteLine($"StatusCode: {response.StatusCode}");
+                context.OutputHelper.WriteLine($"ErrorMessage: {response.ErrorMessage}");
+                Assert.True(response.IsSuccessful, "GET /registry/v2/application failed!");
+            }
+
+            dynamic result = JsonConvert.DeserializeObject<ExpandoObject>(response.Content, new ExpandoObjectConverter());
+            var json = (IDictionary<string, object>)result;
+
+            Assert.True(HasProperty(result, "items"), "GET /registry/v2/application response did not contain items");
+            Assert.False(result.items == null, "GET /registry/v2/application response items property is null");
+
+            foreach (var item in result.items) {
+                var itemDictionary = (IDictionary<string, object>)item;
+
+                if (!itemDictionary.ContainsKey("discoveryUrls")
+                    || !itemDictionary.ContainsKey("applicationId")) {
+                    continue;
+                }
+
+                var discoveryUrls = (List<object>)item.discoveryUrls;
+                var itemUrl = (string)discoveryUrls?.FirstOrDefault(url => IsUrlStringsEqual(url as string, context.DiscoveryUrl));
+
+                if (itemUrl != null) {
+                    return item.applicationId;
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Unregisters a server identified by <paramref name="applicationId"/>
+        /// </summary>
+        /// <param name="context">Shared Context for E2E testing Industrial IoT Platform</param>
+        /// <param name="applicationId">Application ID identifying the server to unregister</param>
+        /// <param name="ct">Cancellation token</param>
+        public static async Task Registry_UnregisterServerAsync(
```

Comment:
this method does not seem symmetrical with the Registry_RegisterServerAsync(). I think it should be.